src/views/view.c

```diff
@@ -1692,6 +1692,25 @@ void dt_view_audio_stop(dt_view_manager_t *vm)
   vm->audio.audio_player_id = -1;
 }
 
+static void _match_rotation(cairo_t *cr, dt_dev_zoom_pos_t d, double dx, double dy,
+                                                              double cx, double cy)
+{
+  const double xx = d[2] - d[0], yx = d[3] - d[1], xy = d[4] - d[0], yy = d[5] - d[1];
+
+  if(fabs(xx * yy - xy * yx) < 1e-10)
+    return;
+
+  double sx = sqrt(xx * xx + yx * yx), sy = sqrt(xy * xy + yy * yy);
```

Comment:
1. wouldn't these been const ?
2. why not use `hypot()` or add and use a `static inline double dt_fast_hypot(const double x,
                                   const double y)` variant ?